lib/trino-parquet/src/main/java/io/trino/parquet/reader/MetadataReader.java

```diff
@@ -80,6 +93,13 @@ public static ParquetMetadata readFooter(ParquetDataSource dataSource, Optional<
                 metadataIndex);
 
         int completeFooterSize = metadataLength + POST_SCRIPT_SIZE;
+        if (maxFooterReadSize.isPresent() && completeFooterSize > maxFooterReadSize.get().toBytes()) {
+            throw new ParquetCorruptionException(
```

Comment:
nit: Is it possible to assert this behaviour by creating a ParquetReader which has very small value ?